vector/src/main/java/im/vector/activity/VectorRoomActivity.java

```diff
@@ -2030,13 +2040,9 @@ private void sendMediasIntent(final Intent intent) {
 
         if (null != intent) {
             sharedDataItems = new ArrayList<>(RoomMediaMessage.listRoomMediaMessages(intent, RoomMediaMessage.class.getClassLoader()));
-        }
-
-        if (null != mLatestTakePictureCameraUri) {
-            if (0 == sharedDataItems.size()) {
-                sharedDataItems.add(new RoomMediaMessage(Uri.parse(mLatestTakePictureCameraUri)));
-            }
-            mLatestTakePictureCameraUri = null;
+            intent.setClass(this, MediaPreviewerActivity.class);
+        } else {
+            intent = new Intent(this, MediaPreviewerActivity.class);
```

Comment:
Ok, thanks, it's clearer now. Maybe you should keep the if to initiate `sharedDataItems`, but move the code related to MediaPreviewerActivity to the if block with `PreferencesManager.previewMediaWhenSending(this)`.